src/pip/_internal/basecommand.py

```diff
@@ -246,12 +242,12 @@ def main(self, args):
             return UNKNOWN_ERROR
         finally:
             # Check if we're using the latest version of pip available
-            if (not options.disable_pip_version_check and not
-                    getattr(options, "no_index", False)):
+            if not options.disable_pip_version_check and not getattr(
+                options, "no_index", False
+            ):
                 with self._build_session(
-                        options,
-                        retries=0,
-                        timeout=min(5, options.timeout)) as session:
+                    options, retries=0, timeout=min(5, options.timeout)
+                ) as session:
```

Comment:
The rules for black are basically:

* If it fits on one line, put it on one line.
* If it doesn't fit on one line, and it's not a data structure or import statement, keep the "internal" items (e.g. arguments) on a single line, and put the opening/closing characters on their own line.
* If that still doesn't fit, put each item on it's own line.

For data structures/import statements, as soon as it can't fit onto one line it explodes them to one line per entry.

It consistently follows this rule for sub items as well.

Reply:
At this point I want to say, can we document the rules we're working to? Blacks documentation says what black *does*, but not what constraints it's applying, so it tends to force you to write code then run black to see if you got it right. There's a lot of "you may have noticed" in the black docs :-(

But I won't say that. At this point, I don't think either of us is going to convince the other. I don't like a lot of the details of black's style, and I prefer to allow programmers some judgement in layout. You're happy with saying "run black on your code and accept the output". Let's just agree to differ, and then it comes down to counting votes (and while I expect my vote to count, and I'd like -1 to count more strongly than +0, I don't feel my opinion should be given extra weight just because I was emphatic in my arguments).

What I *will* do is revise my position to a definite -1, both on the PR as a whole, and also on a number of the style rules black applies.

Reply:
@pfmoore Oh yea, while I have been advocating for the PR, I fully respect your view point here! 

I've been largely viewing it as a difference in opinion, but making an attempt to see if there's some middle ground where there's some specific thing(s) that can be adjusted that makes you feel better about the change. It's perfectly OK if there isn't, and that is perfectly valid. Where I've responded to specific issues you've pointed out, I *hope* that wasn't coming across as "your issue is invalid", but rather as an attempt to give some background into my understanding of *why* black does what it does in that case. There are a few cases where I don't fully agree with black, but I liked at least knowing the rationale behind why black does what it does, because it made it feel less arbitrary.

For what it's worth, I totally get where you're coming from too. Knowing what I know about your workflow-- you're basically not going to get much, if any, of the benefit of the PR. You're just going to get all of the cost of a strict coding style standard that you're not super thrilled with, without the pay-off of "and now you don't have to think about formatting anymore", which more or less requires using some form of integration (even if that's just having black installed and running ``black .``) and in fact, for you it effectively becomes "and now you have to think about formatting even more than before".

On stuff like this, I don't really like pure vote counting, because I feel like everyone should be at least a little bit on board with the idea, even if they're not like, completely enthused. This isn't some weird feature or sub command that you can mostly ignore or something like that, it's pretty foundational to interacting with the project.

With that in mind, I'm going to withdraw this PR, because I'm not comfortable making a change like this when one of the active core devs is a -1 on it and I believe that we've explored the idea to the point where we can confidently say that it's not edge case nits that are the sticking point, but sort of the core of the idea.